config/source_input_services.go

```diff
@@ -0,0 +1,100 @@
+package config
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const serviceSourceInputType = "services"
+
+var _ SourceInputConfig = (*ServicesSourceInputConfig)(nil)
```

Comment:
This file is almost identical to [`config/condition_services.go`](https://github.com/hashicorp/consul-terraform-sync/blob/master/config/condition_services.go). Are there approaches that can be taken to reduce code redundancy? Getting this early would help when catalog services, KV, and additional source input are also supported.